hoomd/hpmc/tune/boxmc_move_size.py

```diff
@@ -0,0 +1,302 @@
+                OnlyFrom(_MoveSizeTuneDefinition.acceptable_attrs,
+                         postprocess=self._flag_new_tunables)
+            ],
+            max_move_size=OnlyIf(
+                to_type_converter({
+                    attr: OnlyTypes(float,
+                                    allow_none=True,
+                                    postprocess=self._flag_move_size_update)
+                    for attr in _MoveSizeTuneDefinition.acceptable_attrs
+                }),))
+        params["boxmc"] = boxmc
+        params["moves"] = moves
+        if max_move_size is None:
+            max_move_size = {
+                attr: None for attr in _MoveSizeTuneDefinition.acceptable_attrs
+            }
+        params["max_move_size"] = max_move_size
+        self._param_dict.update(params)
+
+        self._update_tunables()
+
+    def attach(self, simulation):
+        if not isinstance(simulation.operations.integrator, HPMCIntegrator):
+            raise RuntimeError(
+                "MoveSizeTuner can only be used in HPMC simulations.")
+        super().attach(simulation)
+
+    def act(self, timestep=None):
+        """Tune move sizes.
+
+        Args:
+            timestep (`int`, optional): Current simulation timestep.
+        """
+        # update maximum move sizes
+        if self._should_update_tunables:
+            self._update_tunables()
+        if self._update_move_sizes:
+            for tunable in self._tunables:
+                max_move_size = self.max_move_size[tunable.attr]
+                tunable.domain = (self._min_move_size, max_move_size)
+        super().act(timestep)
+
+    def _update_tunables(self):
+        tunables = self._tunables
+
+        def filter_tunables(tunable):
+            return tunable.attr in self.moves
+
+        self._tunables = list(filter(filter_tunables, tunables))
+        tune_definitions = {t.attr for t in self._tunables}
+
+        # Add any move size tune definitions that are required by the new
+        # specification.
+        for move in self.moves:
+            if move in tune_definitions:
+                continue
+            max_move_size = self.max_move_size[move]
+            move_definition = _MoveSizeTuneDefinition(
+                self.boxmc, move, self.target,
+                (self._min_move_size, max_move_size))
+            self._tunables.append(move_definition)
+        self._should_update_tunables = False
+        self._tuned = 0
+
+    def _flag_move_size_update(self, value):
+        self._update_move_sizes = True
+        return value
+
+    def _flag_new_tunables(self, value):
+        self._should_update_tunables = True
+        return value
+        self._param_dict._cpp_obj = 5
+
+
+class BoxMCMoveSize(_InternalCustomTuner):
+    """Tunes ``BoxMC`` move sizes to targeted acceptance rate.
+
+    For most common creation of a `BoxMCMoveSize` tuner see
+    `BoxMCMoveSize.secant_solver` and `BoxMCMoveSize.scale_solver`
+    respectively.
```

Comment:
Add a full description of the move size tuner including mathematical definitions of what steps it takes, whether the tuner operates indefinitely or stops, how to check if it has tuned, etc... Please do the same for the MC move size tuner. Follow the documentation style in #1237 and #1243. I will avoid modifying the tuner documentation in #1243.

Reply:
Done in `hoomd/tune/attr_tune.py` as it should be.